staging/src/k8s.io/apimachinery/pkg/apis/meta/internalversion/validation/validation.go

```diff
@@ -42,5 +42,28 @@ func ValidateListOptions(options *internalversion.ListOptions) field.ErrorList {
 			allErrs = append(allErrs, field.Forbidden(field.NewPath("resourceVersionMatch"), "resourceVersionMatch \"exact\" is forbidden for resourceVersion \"0\""))
 		}
 	}
+	if options.SendInitialEvents {
+		allErrs = append(allErrs, field.Forbidden(field.NewPath("sendInitialEvents"), "sendInitialEvents is forbidden for list"))
+	}
+	return allErrs
+}
+
+func validateWatchOptions(options *internalversion.ListOptions, isWatchListFeatureEnabled bool) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if match := options.ResourceVersionMatch; len(match) > 0 || options.SendInitialEvents {
+		if len(match) == 0 {
+			allErrs = append(allErrs, field.Forbidden(field.NewPath("sendInitialEvents"), "sendInitialEvents is forbidden for watch unless resourceVersionMatch is provided"))
+			return allErrs
```

Comment:
You don't want to return here, because we effectively want to return all validation errors as part of validation. Same below.